salt/states/pkg.py

```diff
@@ -2344,15 +2348,7 @@ def latest(
                     'comment': 'Invalidly formatted "pkgs" parameter. See '
                                'minion log.'}
     else:
-        if isinstance(pkgs, list) and len(pkgs) == 0:
-            return {
-                'name': name,
-                'changes': {},
-                'result': True,
-                'comment': 'No packages to install provided'
-            }
-        else:
-            desired_pkgs = [name]
+        desired_pkgs = [name]
```

Comment:
Again, this doesn't make sense. Invoking the state with empty `pkgs` is not a supported use, we should be returning the error dict here.

Reply:
It covers the following case
`salt-call state.high {'strace':{'pkg':['latest']}}`  where `pkgs` is `None`